contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -1639,3 +1651,43 @@ pltsql_replace_non_determinstic(text *src_text, text *from_text, text *to_text,
 	}
 	return false;
 }
+
+void
+tsql_set_db_collation(Oid database_collation_oid)
+{
+	/* Initialise collation callbacks */
+	init_and_check_collation_callbacks();
+
+	(*collation_callbacks_ptr->set_db_collation) (database_collation_oid);
+	return;
+}
+
+void
+set_db_collation_internal(int16 db_id)
+{
+	HeapTuple	tuple_sysdb;
+	Form_sysdatabases sysdb;
+	Oid database_collation_oid;
+
+	Oid nspoid = get_namespace_oid("sys", false);
+
+	tuple_sysdb = SearchSysCache1(SYSDATABASEOID, Int16GetDatum(db_id));
+
+	if (!HeapTupleIsValid(tuple_sysdb))
+		return;
+
+	sysdb = ((Form_sysdatabases) GETSTRUCT(tuple_sysdb));
+
+	database_collation_oid = GetSysCacheOid3(COLLNAMEENCNSP,
+											Anum_pg_collation_oid,
+											NameGetDatum(&sysdb->default_collation),
+											Int32GetDatum(-1),
+											ObjectIdGetDatum(nspoid));
+
+	ReleaseSysCache(tuple_sysdb);
+	tsql_set_db_collation(database_collation_oid);
+	is_new_db = true;
+	tsql_get_database_or_server_collation_oid_internal(false);
+	is_new_db = false;
+}
```

Comment:
Lets move this change to common extension. And, we can get rid of this is_new_db variable by invalidating local cache